cmd/mend/mend.go

```diff
@@ -128,12 +187,13 @@ func mend(cmd *cobra.Command, files []string) {
 					}
 					slog.Info("truncated file", "file", filepath, "at", result.truncateAt)
 					stats.truncatedFiles++
-					stats.totalBytesSaved += result.fileSize - result.truncateAt
+					stats.totalBytesTruncated += result.fileSize - result.truncateAt
 				}
 			}
 
 			if result.needsRename {
-				if confirmAction(fmt.Sprintf("remove .open suffix from %s? [y/N] ", filepath), autoYes) {
+				// Auto-rename if no truncation was needed (file is valid, just needs .open suffix removed)
+				if !result.needsTruncate || confirmAction(fmt.Sprintf("remove .open suffix from %s? [y/N] ", filepath), autoYes) {
```

Comment:
instead of relying on past condition for actions I would use a bool state like `wasTruncated`
pure opinionated comment